packages/studio-base/src/providers/LayoutManagerProvider.test.tsx

```diff
@@ -0,0 +1,217 @@
+/** @jest-environment jsdom */
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import { render, waitFor } from "@testing-library/react";
+import { useNetworkState } from "react-use";
+
+import { useVisibilityState } from "@foxglove/hooks";
+import { LayoutData } from "@foxglove/studio-base/context/CurrentLayoutContext";
+import { useLayoutStorage } from "@foxglove/studio-base/context/LayoutStorageContext";
+import { useRemoteLayoutStorage } from "@foxglove/studio-base/context/RemoteLayoutStorageContext";
+import LayoutManagerProvider from "@foxglove/studio-base/providers/LayoutManagerProvider";
+import { LayoutLoader } from "@foxglove/studio-base/services/ILayoutLoader";
+import MockLayoutManager from "@foxglove/studio-base/services/LayoutManager/MockLayoutManager";
+
+// Mock dependencies
+jest.mock("react-use");
+jest.mock("@foxglove/hooks");
+jest.mock("@foxglove/studio-base/context/LayoutStorageContext");
+jest.mock("@foxglove/studio-base/context/RemoteLayoutStorageContext");
+
+const mockLayoutManager = new MockLayoutManager();
+
+jest.mock("@foxglove/studio-base/services/LayoutManager/LayoutManager", () =>
+  jest.fn(() => mockLayoutManager),
+);
+
+describe("LayoutManagerProvider", () => {
+  const mockLayoutLoader: jest.Mocked<LayoutLoader> = {
+    fetchLayouts: jest.fn(),
+    namespace: "local",
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    (useNetworkState as jest.Mock).mockReturnValue({ online: true });
+    (useVisibilityState as jest.Mock).mockReturnValue("visible");
+    (useLayoutStorage as jest.Mock).mockReturnValue({});
```

Comment:
Don't you think it's better to initialize those objects only when needed?
useVisibilityState is used only on one test, for instance. They are being initialized before each test and not being used everytime.

Reply:
I did it because the jest.clearAllMocks( )`, but then I realize the this method only clear the data, not the implementation, so you are correct, it can be mocked only one time. Already moved it